### DISCUSSION TITLE
One Applicaiton Server, Multiple Databases

### DISCUSSION THREAD

**Discussion by luxunxiansheng** [dcm4che/dcm4chee-arc-light#3230] [Q&A]
Hi, there, 
      I check the WIKI, but not sure how to config the system to get this done:
 
Suppose I have A hospital and B hospital, I wolud like both of them have their own database but share one application server.   That is,  I could easily unmount  A or B  in a clean way if necessay.   Also,  If possible, I could easily mount a new hospital say, C, without touching  the original A and B databases . 

     Thanks in advance. 

L.B.

**Comment by gunterze** [comment]
You would have to customize the build to build two different EARs,  referring different Data Sources:
- `-Dds=java:/PacsDS_A` 
- `-Dds=java:/PacsDS_B` 

and specifying different [context Roots for included web application](https://github.com/dcm4che/dcm4chee-arc-light/blob/master/dcm4chee-arc-ear/pom.xml#L522-L548).

But it's not tested and so I can't give any support.

Last 15 years, people moved away from bundling multiple deployments into one application server to separating services running in separate containers.

**Reply by luxunxiansheng** [reply]
Hi, Gunterze,
      Let me narrow our case and there might be better solution from you side. 
      We may host more than 100 different size of  hospitals in cloud.  For those big ones, there will be 3000 studies per day while for those small ones, dozens of studies per day.   As mentioned , we prefer that each hospital has  its own  database  for  some sort of  data  sercurity policy and in case to add or remove one hospita for some commercial reason in the future.   But  if each has its application server,  the cost will be a problem.   
     I am not sure whether what  I proposed in my last post is proper or not. Any help will be appreciated. 

L.B.

**Comment by gunterze** [comment]
What "costs" you refer for running one archive container for each hospital?

**Reply by luxunxiansheng** [reply]
I am not sure I fully understood your question.  I think we have to setup the CPU/Mem  resources to run the application server for each hospital if not  "customize the build to build two different EARs, referring different Data Sources" as you suggested.

**Reply by luxunxiansheng** [reply]
Or let me change my quesiton as this:
      Given my case, say,  data security policy and easily to add or remove one hospital ,  what is the cost effective way to  config dcm4chee .

**Reply by gunterze** [reply]
Most relevant cost for archive containers are the allocated memory - which you may tailor according the expected load for each particular instance. Assuming 500MB minimum, you will need 50GB minimum for running 100 archive containers.

**Comment by gunterze** [comment]
BTW, using one DB for multiple hospital, you still can isolate the archive content for each hospital by using hospital specific Issuer of Patient IDs and/or  Universal Entity IDs for Patient IDs and configuring [Access control based on Archive AEs](https://github.com/dcm4che/dcm4chee-arc-light/wiki/Access-control-based-on-Archive-AEs)

**Reply by luxunxiansheng** [reply]
Thank you very much, Gunterze.   We team will take a discussion and may come back for more helps from you .
